broker_test.go

```diff
@@ -1,31 +1,388 @@
 package main
 
-import "testing"
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"testing"
 
-func TestBroker_Services(t *testing.T) {
-	t.Skip("pending")
+	"github.com/hashicorp/vault/api"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+type Environment struct {
+	Context          context.Context
+	Broker           *Broker
+	InstanceID       string
+	BindingID        string
+	SpaceGUID        string
+	OrganizationGUID string
+	Async            bool
+}
+
+func TestBroker(t *testing.T) {
```

Comment:
I am not adverse to doing that! :-)

The reason I take this approach is actually to give very specific names for each test. It's originally inspired by [this post](https://testing.googleblog.com/2007/02/tott-naming-unit-tests-responsibly.html) from Google's Testing on the Toilet which I think is a funny but also useful set of blog posts. The idea is that it's easy for a maintainer to very quickly zoom to the problem upon viewing a test failure, from the test name alone.

That said, the linked post is for unit tests, not functional tests, and it would be trivial for me to switch them over so I will. Plus I have no strong opinion, just wanted to share the reasoning.

Reply:
Pushed in https://github.com/hashicorp/vault-service-broker/pull/23/commits/13368200d7132fb8974918d54baab5f6d5fb42a7

Also, in switching it over, I can see how it would be easier to reason about stateless tests in a different kind of way.

Reply:
Looks good.  Yes, good test names are great, though I admit to often having trouble capturing the nature of a complex test in a single func name - especially when there are multiple variants of the test with subtle differences.